api/src/main/java/org/openmrs/module/printer/PrinterServiceImpl.java

```diff
@@ -145,12 +149,11 @@ public void setDefaultPrinter(Location location, PrinterType type, Printer print
 
         LocationAttributeType locationAttributeType = getLocationAttributeTypeDefaultPrinter(type);
 
-        // if no printer is specified, void any existing default printer
-        if (printer == null) {
-            for (LocationAttribute attr : location.getActiveAttributes(locationAttributeType)) {
-                attr.setVoided(true);
-            }
-        } else {
+        for (LocationAttribute attr : location.getActiveAttributes(locationAttributeType)) {
+            location.getAttributes().remove(attr);
```

Comment:
I changed this to just remove (hard delete) the attribute rather than void it (and did the same below).  at th end of the day, I don't think this is what was causing the bug, but seems unnecessary to audit this data.